src/models/models.cpp

```diff
@@ -14,10 +14,25 @@
 
 namespace xft {
 Model::~Model() {
+    exitSlaves();
     if (decoder != nullptr) { delete decoder; }
     if (searcher != nullptr) { delete searcher; }
 }
 
+void Model::exitSlaves() {
+    if (decoder->getRank() == 0) {
+        configuration.maxLen = 0;
+        configuration.numBeams = 0;
+        configuration.numBeamHypsToKeep = 0;
+        configuration.lenPenalty = 0;
+        configuration.doEarlyStopping = true;
+        configuration.eosTokenId = 0;
+        configuration.padTokenId = 0;
```

Comment:
Just set numBeams == 0 is ok. Some other settings will be added and it will redundant if set all of attributes.